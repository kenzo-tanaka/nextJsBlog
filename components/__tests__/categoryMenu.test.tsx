import React from "react";
import { createRenderer } from "react-test-renderer/shallow";
import CategoryMenu from "../categoryMenu";

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/",
      pathname: "",
      query: "",
      asPath: "",
    };
  },
}));

const renderer = createRenderer();
test("CategoryMenu", () => {
  renderer.render(<CategoryMenu />);
  const renderedOutput = renderer.getRenderOutput();
  expect(renderedOutput).toMatchSnapshot();
});
