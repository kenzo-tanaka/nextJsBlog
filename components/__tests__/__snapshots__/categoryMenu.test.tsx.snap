// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryMenu 1`] = `
<div>
  <a
    className="p-2 font-medium text-gray-800 w-full inline-block hover:bg-gray-100 hover:no-underline "
    href="/"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    All
  </a>
  <a
    className="p-2 font-medium text-gray-800 w-full inline-block hover:bg-gray-100 hover:no-underline "
    href="/categories/dev"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    dev
  </a>
  <a
    className="p-2 font-medium text-gray-800 w-full inline-block hover:bg-gray-100 hover:no-underline "
    href="/categories/work"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    work
  </a>
</div>
`;
